test/test-0018.sh

```diff
@@ -5,34 +5,38 @@
 cleanup 18
 
 # ------------------------------- Test 18 ------------------------------------
-preptest test.log 18 1
+needle="$(uuidgen -r)"
```

Comment:
On my system ```uuidgen``` is installed by the util-linux package?  Is it available on non-linux platforms (e.g. cygwin)?  If not, we should not use it unconditionally.  Why should any files survive across test runs?  Is not the proper fix to just clean files properly before running the test?

Reply:
> How are you confirming that journalctl is unconditionally used?

Not exactly. Lack of journalctl causes SYSLOG_TESTS value to remain zero, and the finding a message part of the test is not done. In effect, if I understand the idea of this test correctly, the logrotate syslogging is only verified on systems that has working journal. Other systems simply skip the verification, which is not great but we can perhaps tolerate that - at least the check can work somewhere. But as I mentioned earlier journal entries are global, that is not session or user specific, one has to be careful not to mix a check payload from one run with an another. Hence the change I wrote.

Reply:
OK.  So unless you want to split the test into two separate tests (one generic and one Linux specific), the following patch should avoid the test-suite failure on systems without ```uuidgen```:
```
--- a/test/test-0018.sh
+++ b/test/test-0018.sh
@@ -2,6 +2,9 @@

 . ./test-common.sh

+# uuidgen is required by this test, availability of journalctl is optional
+uuidgen --version || exit 77
+
 cleanup 18

 # ------------------------------- Test 18 ------------------------------------
```
Are you fine with this fixup?

Reply:
@kerolasa ping?

Reply:
This seems to get staled.  I will push everything except this commit to move things forward.  Please open a separate pull request for the changes to ```test-0018.sh``` if you still want to get them in...